schema/changelog-3.10.xml

```diff
@@ -62,17 +62,51 @@
     </createTable>
 
     <addForeignKeyConstraint baseTableName="user_user" baseColumnNames="userid" constraintName="fk_user_user_userid" referencedTableName="users" referencedColumnNames="id" onDelete="CASCADE" />
-    <addForeignKeyConstraint baseTableName="user_user" baseColumnNames="manageduserid" constraintName="fk_user_user_manageduserid" referencedTableName="users" referencedColumnNames="id" onDelete="CASCADE" />
 
     <update tableName="users">
         <column name="devicelimit" valueNumeric="-1" />
         <where>devicelimit = 0</where>
     </update>
+    <dropDefaultValue tableName="users" columnName="devicelimit" />
     <addDefaultValue tableName="users" columnName="devicelimit" defaultValueNumeric="-1" />
 
     <addColumn tableName="users">
       <column name="devicereadonly" type="BOOLEAN" defaultValueBoolean="false" />
     </addColumn>
 
   </changeSet>
+
+  <changeSet author="author" id="changelog-3.10-notmssql">
+
+    <preConditions onFail="MARK_RAN">
+      <not>
+        <dbms type="mssql" />
+      </not>
+    </preConditions>
+    <addForeignKeyConstraint baseTableName="user_user" baseColumnNames="manageduserid" constraintName="fk_user_user_manageduserid" referencedTableName="users" referencedColumnNames="id" onDelete="CASCADE" />
+
+  </changeSet>
+
+  <changeSet author="author" id="changelog-3.10-mssql">
+
+    <preConditions onFail="MARK_RAN">
+      <dbms type="mssql" />
+    </preConditions>
+    <sql>
+      CREATE TRIGGER tg_users_delete
+      ON users FOR DELETE
+      AS BEGIN
+        DELETE FROM user_user WHERE manageduserid IN (SELECT deleted.id FROM deleted)
+      END
+    </sql>
+    <sql>
+      CREATE TRIGGER tg_groups_delete
```

Comment:
I'm thinking if it's better to split it into `changelog-3.7-mssql`. What do you think?